backend/inspirehep/tools/views.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+from io import BytesIO, TextIOWrapper
+
+from flask import Blueprint, abort, current_app, jsonify, request
+
+from inspirehep.files.api import current_s3_instance
+from inspirehep.records.utils import hash_data
+from inspirehep.tools.utils import get_filename, get_mimetype, get_references
+
+blueprint = Blueprint("inspirehep_tools", __name__, url_prefix="/tools")
+
+ALLOWED_EXTENSIONS = {"tex"}
+
+
+def allowed_file(filename):
+    return "." in filename and filename.rsplit(".", 1)[1].lower() in ALLOWED_EXTENSIONS
+
+
+@blueprint.route("/bibgen", methods=["POST"])
+def generate_bibliography():
+
+    requested_format = request.args.get("format")
+
+    bytes_file = request.files["file"]
```

Comment:
how does this work? is the request streamed? what happens if I send a 20 GB (so larger than the web node memory) file?

Reply:
Can we have such a case? We could save the file and work with it, then upload it to S3 and delete it from the node's storage. Is it worth it?

Reply:
no, I'm not saying we'll have actual tex files containing 20GB of data. All I'm saying is that I can create such a file, slap a `.tex` extension on it and submit it. So you should probably limit file size to some reasonable size (10MB seems plenty, even less should be ok).

Reply:
I limited it to 10MB for this endpoint. Request size is limited to 100MB for the whole application anyway.